### ISSUE TITLE
Create a proper release process

### ISSUE THREAD

**Issue by vimukthi-git** [vimukthi-git/beanstalkg#30]
We need to create a proper release process so that users on a targeted platform can use `beanstalkg` easily. 

### Targeted platforms
1. Container environments(build image as explained in https://docs.docker.com/engine/userguide/eng-image/multistage-build/)
2. rpm packages for the linux variants that use it
3. deb packages for the linux variants that use it

We can probably follow something like https://github.com/goshawkdb/packaging for the linux distros. More research is still needed.

**Comment by vimukthi-git** [comment]
@stephenwithav may be you can do some update for docker file to cover the case 1.

**Comment by stephenwithav** [comment]
The multi-stage builds look fantastic!  Thanks for pointing them out.

**Comment by vimukthi-git** [comment]
@stephenwithav  thanks. I've also created a first release, v0.0.1 so that we can try the release process(release docker image, rpm, deb etc.). I will also create a script to build the artefacts needed.

**Comment by stephenwithav** [comment]
Will the release process push an image to Docker Hub/Store?

**Comment by vimukthi-git** [comment]
Yes that is the target as well as following semantic versioning. Whole process needs to be automated in CircleCi project I have for beanstalkg, so I will handle the next few parts for this ticket. You will review of course.

**Comment by vimukthi-git** [comment]
I have made some progress on how to add a proper circleci config so that we can build the docker images. https://github.com/circleci/cci-demo-go/blob/master/.circleci/config.yml shows a config with circleci 2. We currently use circleci 1 so I want to migrate as well. I will add the sample yml file I came up with soon to test in a pull request. This is a bit more complicated since we need to use multi stage builds as well. But can be done with the way show in https://discuss.circleci.com/t/multi-stage-docker/12386.
